backend/tests/unit/oasst_backend/api/v1/test_tasks.py

```diff
@@ -0,0 +1,69 @@
+import json
+from unittest import mock
+
+import pytest
+from fastapi import FastAPI
+from fastapi.testclient import TestClient
+from oasst_backend.api.v1 import tasks
+from oasst_backend.api.v1.tasks import TaskContext
+from oasst_backend.prompt_repository import PromptRepository
+from oasst_backend.task_repository import TaskRepository
+from oasst_backend.tree_manager import TreeManager
+from oasst_shared.schemas.protocol import TaskDone, TaskRequest
+from starlette.status import HTTP_200_OK, HTTP_422_UNPROCESSABLE_ENTITY
+
+
+class FakeApp:
+    def __init__(self, app: FastAPI):
+        self.app = app
+
+        fake = mock.Mock(spec=TaskContext)
+        fake.pr = mock.Mock(spec=PromptRepository)
+        fake.tm = mock.Mock(spec=TreeManager)
+        fake.tm.next_task.return_value = TaskDone(), "fake_tree_id", "fake_parent_message_id"
+        fake.pr.task_repository = mock.Mock(spec=TaskRepository)
```

Comment:
Be careful about using `spec` with `return_value`... even though you're using spec, you can still set `return_value` to whatever you want. We should also have integration tests, so we can catch this sort of issue

Reply:
Removed specs, and restructured the test a bit.